catch/hipTestMain/main.cc

```diff
@@ -1,16 +1,50 @@
 #define CATCH_CONFIG_RUNNER
+#include <cmd_options.hh>
 #include <hip_test_common.hh>
 #include <iostream>
 
+CmdOptions cmd_options;
+
 int main(int argc, char** argv) {
   auto& context = TestContext::get(argc, argv);
   if (context.skipTest()) {
     // CTest uses this regex to figure out if the test has been skipped
     std::cout << "HIP_SKIP_THIS_TEST" << std::endl;
     return 0;
   }
-  int out = Catch::Session().run(argc, argv);
+
+  Catch::Session session;
+
+  using namespace Catch::clara;
+  // clang-format off
+  auto cli = session.cli() 
+    | Opt(cmd_options.iterations, "iterations")
+        ["-I"]["--iterations"]
+        ("Number of iterations used for performance tests (default: 1000)")
+    | Opt(cmd_options.warmups, "warmups")
+        ["-W"]["--warmups"]
+        ("Number of warmup iterations used for performance tests (default: 100)")
+    | Opt(cmd_options.no_display)
+        ["-S"]["--no-display"]
+        ("Do not display the output of performance tests")
+    | Opt(cmd_options.progress)
+        ["-P"]["--progress"]
+        ("Show progress bar when running performance tests")
+    | Opt(cmd_options.cg_extended_run, "cg_extened_run")
+        ["-E"]["--cg-extended-run"]
+        ("Number of iterations used for cooperative groups sync tests (default: 5)")
+    | Opt(cmd_options.accuracy_iterations, "accuracy_iterations")
+        ["-A"]["--accuracy-iterations"]
+        ("Number of iterations used for math accuracy tests with randomly generated inputs (default: 2^32)")
```

Comment:
The actual number of iterations should probably depend on the type/precision and the arity of the function being tested.  This is a maximum, I expect.

Reply:
There are three types of tests based on the input values used: special case, random, and all possible input values. The type of test used does depend on the arity and precision of the function. For example, unary single-precision functions are tested for all possible input values, whereas binary or double-precision functions are tested with a number of random values and a table of special values. The --accuracy-iterations option only controls the number of random input values we generate for the random input tests.

Reply:
But what happens if, e.g. a value of more than 65536 is given for a unary f16 function?  Will it reduce the number, or will it use the full number (wasting time).